Automattic/wp-calypso
# Checkout: Don't use localStorage to save product data for the thank you page

Fixes #2235.

This PR updates the checkout process, and:
- Changes the route for the checkout thank you page from `/checkout/thank-you` to `/checkout/:site_slug/:receipt_id/thank-you`.
- Adds actions/a reducer/selector to fetch a receipt from the server and store it in the global state tree.
- Dispatches an action to store a receipt when a transaction is completed successfully.
- Fetches a receipt on the thank you page.
- Updates the thank you page to render using purchases from the receipt, instead of items from the last cart.
- Removes `lastTransaction` from local storage, and any logic that used to use that property.

#### Testing instructions - Credit card

1. Run `git checkout fix/thank-you-page` and start your server
2. Open the [`Plans` page](http://calypso.localhost:3000/plans/:site)
3. Purchase a plan with a credit credit card
4. Assert that you are redirect to the thank you page with the correct information on it
5. Navigate to another page (like reader)
6. Assert that clicking back reloads the thank you page

#### Testing instructions - PayPal

1. Run `git checkout fix/thank-you-page` and start your server
2. Open the [`Plans` page](http://calypso.localhost:3000/plans/:site)
3. Purchase a plan with paypal
4. Go through the paypal flow
4. Assert that you are redirect to the thank you page with the correct information on it
5. Navigate to another page (like reader)
6. Assert that clicking back reloads the thank you page

#### Testing instructions - generic thank you page
1. Visit `/checkout/:site/thank-you` for any site.
2. Assert that you see a [generic thank you page](https://cloudup.com/cpStlBbu7LJ).

#### Additional notes

Also worth testing:
- different plans
- premium themes
- domains

#### Reviews
